packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -287,12 +369,30 @@ export function createInsightsMiddleware<
             queuedUserToken,
             queuedAuthenticatedUserToken
           );
+        } else if (anonymousUserToken) {
+          setUserToken(anonymousUserToken, anonymousUserToken, undefined);
```

Comment:
would like to see the setUserToken function simplified (can be separate pr) to not have three arguments